### ISSUE TITLE
EntityAdapter generic methods improvement suggestion

### ISSUE THREAD

**Issue by Arsync** [ForeverFast/ForeverFast.Fluxor.Extensions#1]
Generic methods parameter 'state' of type `EntityState<TKey, TEntity>` can be simplified

```cs
public TState Remove<TState>(TKey id, EntityState<TKey, TEntity> state)
    where TState : EntityState<TKey, TEntity>
    => (TState)state with
    {
        Entities = state.Entities.Remove(id),
    };

```
Simplified version:
```cs
public TState Remove<TState>(TKey id, TState state)
    where TState : EntityState<TKey, TEntity>
    => state with
    {
        Entities = state.Entities.Remove(id),
    };
```
It allows to know return type from parameter 'state' and call adapter actions without generic type argument:
`Adapter.Remove(entity.Id, state)`

**Comment by Metalvast** [comment]
I agree, this will greatly simplify interaction with EntityAdapter. I will try to do this this week and remove the dependency on AdapterEntity.

**Comment by Metalvast** [comment]
Well... it was fast. 
Fixed in v0.0.3.
